src/Config.php

```diff
@@ -0,0 +1,343 @@
+<?php namespace Rollbar;
+
+use Rollbar\Payload\Payload;
+use Rollbar\Payload\Level;
+
+if (!defined('ROLLBAR_INCLUDED_ERRNO_BITMASK')) {
+    define(
+        'ROLLBAR_INCLUDED_ERRNO_BITMASK',
+        E_ERROR | E_WARNING | E_PARSE | E_CORE_ERROR | E_USER_ERROR | E_RECOVERABLE_ERROR
+    );
+}
+
+class Config
+{
+    private $accessToken;
+    /**
+     * @var DataBuilder
+     */
+    private $dataBuilder;
+    private $configArray;
+    /**
+     * @var TransformerInterface
+     */
+    private $transformer;
+    /**
+     * @var FilterInterface
+     */
+    private $filter;
+    private $minimumLevel;
+    /**
+     * @var ResponseHandlerInterface
+     */
+    private $responseHandler;
+    /**
+     * @var \Rollbar\Senders\SenderInterface
+     */
+    private $sender;
+    private $reportSuppressed;
+    /**
+     * @var callable
+     */
+    private $checkIgnore;
+    private $error_sample_rates = array();
+    private $mt_randmax;
+
+    private $included_errno = ROLLBAR_INCLUDED_ERRNO_BITMASK;
+
+    public function __construct(array $configArray)
+    {
+        $this->updateConfig($configArray);
+
+        if (isset($configArray['error_sample_rates'])) {
+            $this->error_sample_rates = $configArray['error_sample_rates'];
+        }
+
+        $levels = array(E_WARNING, E_NOTICE, E_USER_ERROR, E_USER_WARNING,
+            E_USER_NOTICE, E_STRICT, E_RECOVERABLE_ERROR);
+        // PHP 5.3.0
+        if (defined('E_DEPRECATED')) {
+            $levels = array_merge($levels, array(E_DEPRECATED, E_USER_DEPRECATED));
+        }
+        $curr = 1;
+        for ($i = 0, $num = count($levels); $i < $num; $i++) {
+            $level = $levels[$i];
+            if (!isset($this->error_sample_rates[$level])) {
+                $this->error_sample_rates[$level] = $curr;
+            }
+        }
+        $this->mt_randmax = mt_getrandmax();
+    }
+
+    public function configure($config)
+    {
+        $this->updateConfig($this->extend($config));
+    }
+
+    public function extend($config)
+    {
+        return array_replace_recursive(array(), $this->configArray, $config);
+    }
+
+    public function getConfigArray()
+    {
+        return $this->configArray;
+    }
+
+    protected function updateConfig($c)
+    {
+        $this->configArray = $c;
+        
+        $this->setAccessToken($c);
+        $this->setDataBuilder($c);
+        $this->setTransformer($c);
+        $this->setMinimumLevel($c);
+        $this->setReportSuppressed($c);
+        $this->setFilters($c);
+        $this->setSender($c);
+        $this->setResponseHandler($c);
+        $this->setCheckIgnoreFunction($c);
+
+        if (isset($c['included_errno'])) {
+            $this->included_errno = $c['included_errno'];
+        }
+    }
+
+    private function setAccessToken($c)
+    {
+        if (isset($_ENV['ROLLBAR_ACCESS_TOKEN']) && !isset($config['access_token'])) {
+            $config['access_token'] = $_ENV['ROLLBAR_ACCESS_TOKEN'];
```

Comment:
Looks like a little bit of variable confusion here. Param in the function is `$c`, but here and in the conditional we're checking for `$config`.